registry/handlers/manifests.go

```diff
@@ -481,12 +483,20 @@ func (imh *manifestHandler) DeleteManifest(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
+	g := errgroup.Group{}
+	g.SetLimit(storage.DefaultConcurrencyLimit)
 	for _, tag := range referencedTags {
-		if err := tagService.Untag(imh, tag); err != nil {
-			imh.Errors = append(imh.Errors, err)
-			return
-		}
+		tag := tag
+
+		g.Go(func() error {
+			if err := tagService.Untag(imh, tag); err != nil {
+				imh.Errors = append(imh.Errors, err)
```

Comment:
Also, we need a `mutex` here; we can't be appending `imh.Errors` concurrently without a mutex

Reply:
The mutex doesn't have to be on `imh` (read: I will not approve this PR unless it is changed.)
```go
var (
	errs []error
	mu   sync.Mutex
)
for _, tag := range referencedTags {
	g.Go(func() error {
		if err := tagService.Untag(imh, tag); err != nil {
			mu.Lock()
			errs = append(errs, err)
			mu.Unlock()
		}
		return nil
	})
}
_ = g.Wait()
imh.Errors = errs
```